packages/@expo/cli/src/start/platforms/PlatformManager.ts

```diff
@@ -136,7 +163,22 @@ export class PlatformManager<
       resolveSettings.device
     );
     if (options.runtime === 'expo') {
-      return this.openProjectInExpoGoAsync(resolveSettings);
+      const deviceManager = await this.props.resolveDeviceAsync(resolveSettings);
+      if (this.props.getInterstitialPageUrl()) {
+        const applicationId = await this._getAppIdResolver().getAppIdAsync();
+        if (await deviceManager.isAppInstalledAsync(applicationId)) {
+          return this.openProjectInterstitialPageAsync(deviceManager);
+        } else {
+          Log.warn(
+            `\u203A The 'expo-dev-client' package is installed, but a development build isn't ` +
+              `available.\nYour app will open in Expo Go instead. If you want to use a ` +
+              `development build, you need to make and install one first.\n${learnMore(
+                'https://docs.expo.dev/development/build/'
+              )}`
+          );
+        }
+      }
+      return this.openProjectInExpoGoAsync(deviceManager);
```

Comment:
The `openAsync` function currently delegates out complex open options to preserve sanity and keep the code readable, this change seems to extract the existing functionality from `openProjectInExpoGoAsync` and `getExpoGoUrl` into `openAsync`.